Registry - Keybinding - Skill - Config modules

minheragon12345 opened the pull request:
# Description

- Added Registry Module
- Added Keybinding Module
- Added Skill Module
- Added Config Module

## List of changes

- Added Registry Module
- Added Keybinding Module
- Added Skill Module
- Added Config Module

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code Refactoring (non-breaking change which improved the Code Quality)
- [ ] Breaking Code Refactoring (breaking change which improved the Code Quality)
- [ ] Documentation update (JavaDoc or Markdown change)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Full skill system: learnable skills with toggle/hold/scroll controls, per-skill modes, client input handling, and networked activation/release/scroll/toggle.
  * Race system: set/evolve races with abilities, intrinsic skills, evolution progress and networked evolution/ability sync.
  * Attributes & combat: critical-hit mechanics, glide/lava/swim speed modifiers, movement boosts, and expanded projectile/damage hooks.
  * Configs & sync: TOML configs with load/save and automatic server→client sync on join.
  * Commands & keybindings: many new argument types and a client keybinding manager.
  * Testing: sample skills, races, items, keybindings and test utilities.

* **Chores**
  * Build, packaging and version/property updates; localization additions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

maltesermailo committed:
Added files that are necessary for all other modules to work OR are too small to fit into a future module

Erusel committed:
New config Module and Feature with Architectury

Erusel committed:
Added neoforge module

Erusel committed:
Some code cleaning

Erusel committed:
Fixed dumb fail mistake

minheragon12345 committed:
Skill System

minheragon12345 committed:
Merge remote-tracking branch 'origin/1.21.1/master' into 1.21.1/skills

# Conflicts:
#	settings.gradle
#	testing-common/build.gradle

minheragon12345 committed:
Keybinding & Registry

minheragon12345 committed:
Mob Effect Registry

minheragon12345 committed:
Potion Registry

minheragon12345 committed:
Cooldown with modes

minheragon12345 committed:
Merge branch '1.21.1/test-branch' into 1.21.1/master

minheragon12345 committed:
Merge pull request #47 from Erusel/1.21.1/master

Idk how to check out forked branch

minheragon12345 committed:
Merge branch '1.21.1/skills' into 1.21.1/test-branch

# Conflicts:
#	settings.gradle
#	testing-common/build.gradle
#	testing-common/src/main/java/io/github/manasmods/manascore/testing/ManasCoreTesting.java

minheragon12345 committed:
Merged with Conig & Commented out attribute stuff to run Client

minheragon12345 committed:
Parent folder path for Config + more test

minheragon12345 committed:
Revert and fixes + Icons

minheragon12345 committed:
Merge remote-tracking branch 'origin/1.21.1/master' into 1.21.1/minh

# Conflicts:
#	build.gradle

minheragon12345 committed:
Updated copyright year to 2024-2025

minheragon12345 committed:
Fabric works now (when exported)

minheragon12345 committed:
Removed Registry Module and Added Arch's registry for Testing module

minheragon12345 committed:
Fixed Keybind and Skill not working on NeoForge

minheragon12345 committed:
Attribute Module & Attribute Registry

minheragon12345 committed:
Race Module Started

minheragon12345 committed:
Glide/Lava/Swim Speed attributes

minheragon12345 committed:
Respawn Dimension for Races

minheragon12345 committed:
Config Module revamped as Toml file with @Comment

minheragon12345 committed:
Field insertion order & SubConfig loading

minheragon12345 committed:
Config client-server syncing & API Comments for documenting

minheragon12345 committed:
Some code changes following Coderabbit's suggestions & Config Fabric fix

minheragon12345 committed:
More of Coderabbit's suggestions & Fixed Skill/Race attribute modifiers removal not working

minheragon12345 committed:
More Command parameters & Minor commenting changes

minheragon12345 committed:
Code refactoring for Temporary Maven publish

minheragon12345 committed:
Attribute Registry fix

minheragon12345 committed:
Fixed Attribute not applying on modded entities

minheragon12345 committed:
Fabric Modded Entities doesn't get modded attributes

minheragon12345 committed:
Vanilla Mobs dont get any custom attributes.

minheragon12345 committed:
ManasSkill improvement

minheragon12345 committed:
Storages doesn't update owner after cloning/respawning + Manas Skill QOL

minheragon12345 committed:
Race abilities' qol + Minor Skill changes + Glide events

minheragon12345 committed:
Elytra not working fixed + Temporary Fabric Command fix

minheragon12345 committed:
Obtained Intrinsic Skills for races

minheragon12345 committed:
More SpawnPoint/Teleportation Utils

minheragon12345 committed:
Skill Scroll with modes + Modes with instance

minheragon12345 committed:
Race Stat client spam

minheragon12345 committed:
Custom Attribute is overwritten by default value

minheragon12345 committed:
Skill Concurrent fix + Race save attributes leaving End + Skill Mastery Event

minheragon12345 committed:
Skill Sub-Instances

minheragon12345 committed:
Some Event changes

minheragon12345 committed:
Improved some syncing

minheragon12345 committed:
Post Init events

minheragon12345 committed:
TickingSkill Improvement

minheragon12345 committed:
TickingSkill & Target Event fixes

minheragon12345 committed:
Race on target fix

minheragon12345 committed:
Registry Key fix

minheragon12345 committed:
Small fixes

minheragon12345 committed:
Command Perm fixes + More command arguments + Misc changes

Charismara committed:
feat: Add `bundle` modules for publishing

minheragon12345 committed:
Merge pull request #50 from ManasMods/1.21.1/professor

Add `bundle` modules for publishing

TotallyDem committed:
Attempt at diagnosing Invalid Player Data crash causes

minheragon12345 committed:
Merge pull request #51 from TotallyDem/1.21.1/minh

Attempt at diagnosing Invalid Player Data crash causes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/ManasMods/ManasCore?pullRequest=48) <br/>All committers have signed the CLA.

minheragon12345 committed:
Passenger bug + Bundle

minheragon12345 committed:
Version id changed

TotallyDem committed:
Merge pull request #1 from ManasMods/1.21.1/minh

1.21.1/minh

TotallyDem committed:
Addressing failing to register, thought to have been a race condition

Replaced EventFactory in StorageEvents with a new class ManasEventFactory, which uses a ConcurrentLinkedList instead of the ArrayList that EventFactory uses

TotallyDem committed:
Changed ManasEventFactory so it can be invoked more than once.

Previously, since we were using the ConcurrentLinkedQueue as is, the registers would have potentially be un-registered after first use. To prevent that, I used non-destructive methods of getting the listeners from the queue.

minheragon12345 committed:
Merge pull request #52 from TotallyDem/1.21.1/minh

Listener registration fix

minheragon12345 committed:
Cooldown Tag

TotallyDem committed:
Reworked SkillStorage to use an internal flag for cooldown ticking rather than an entity flag. Added documentation for methods. Updated ManasSkillInstance to support this new process. Removed the depricated entity tag.

TotallyDem committed:
Ticking skill rework to let all entities use ticking skills while reducing lag of checking if entities have skills to tick.

minheragon12345 committed:
Merge pull request #53 from TotallyDem/1.21.1/minh

Skills Cooldown/Temporary skill tick rework

minheragon12345 committed:
Cooldown Tag removed

TotallyDem committed:
Manascore-architectury initial commit

Added new module which adds parallel mod loading fixes for the Neoforge version of manascore.

TotallyDem committed:
Updated race held ability to use a TickingRace stored within the storage. Make heldTicks server authoratative.

TotallyDem committed:
Changed Skill heldTicks to be server authoratative.

TotallyDem committed:
Alternative cleaner implementation for MixinEventFactory. Needs verification if it works.

TotallyDem committed:
AI recommended tweaks.

TotallyDem committed:
Removed useless imports, improved initial loop logic for SkillStorage#handleSkillRelease

TotallyDem committed:
Removed redundant code from RequestRaceAbilityActivationPacket

TotallyDem committed:
Moved Skill activation code from RequestSkillActivationPacket to SkillStorage

minheragon12345 committed:
Merge pull request #54 from TotallyDem/1.21.1/minh

Race, Skill, and Arch updates

minheragon12345 committed:
Small changes

minheragon12345 committed:
onRelease didnt work properly

TotallyDem committed:
setCooldowns didn't mark cooldown tick properly.

minheragon12345 committed:
Merge pull request #55 from TotallyDem/1.21.1/minh

setCooldowns didn't mark cooldown tick properly.

minheragon12345 committed:
Version update

minheragon12345 committed:
Config and Sync fix

minheragon12345 committed:
canActivate

minheragon12345 committed:
feat(player-animation): added basic player animation system

minheragon12345 committed:
feat(player-animation): added playThenLoop trigger

minheragon12345 committed:
feat(player-animation): added conditional animations

minheragon12345 committed:
feat(player-animation): added additive and aim bones for animations

minheragon12345 committed:
feat(player-animation): added aim body option

minheragon12345 committed:
feat(player-animation): added held items animation support

minheragon12345 committed:
feat(player-animation): added requires_animation field for animations for better sequencing
